src/hotspot/cpu/riscv/templateTable_riscv_64.cpp

```diff
@@ -240,48 +240,59 @@ void TemplateTable::sipush() {
 void TemplateTable::ldc(bool wide) {
   Register Rscratch1 = R5_scratch1,
            Rscratch2 = R6_scratch2,
-           Rcpool    = R3_ARG1_PPC;
+           Rcpool    = R7_scratch3;
```

Comment:
Using outgoing C arg registers as temps is fine, until you call C functions. There should be no uses of Rcpool after C call. 
Scratch registers r5-r7 are caller save, so there is no benefit of them compared to arg registers -- they will be trashed in C anyway